Backport: Added support for hypervisor_hostname to v2

ngin8-beta opened the pull request:
<!--
Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/main/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

-->
Fixes #3307

EmilienM commented:
Could you please backport that commit: https://github.com/gophercloud/gophercloud/pull/3301/commits/da9417f3358d56bae822ebb6386c70305b336d1c instead of the merge commit. Thanks

ngin8-beta commented:
I have tried cherry-picking da9417f as instructed. However, since this is my first time using the cherry-pick feature, I’m not sure if I did it correctly.

ngin8-beta commented:
```
❯ git reset --hard HEAD~1

HEAD is now at 2e6154e0 Merge pull request #3299 from shiftstack/prepare_v2.5.0
❯ git status
On branch v2
Your branch is up to date with 'upstream/v2'.

nothing to commit, working tree clean
❯ git cherry-pick da9417f
Auto-merging openstack/compute/v2/servers/requests.go
CONFLICT (content): Merge conflict in openstack/compute/v2/servers/requests.go
Auto-merging openstack/compute/v2/servers/testing/requests_test.go
error: could not apply da9417f3... Added support for hypervisor_hostname
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
❯ git status
On branch v2
Your branch is up to date with 'upstream/v2'.

You are currently cherry-picking commit da9417f3.
  (fix conflicts and run "git cherry-pick --continue")
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:
	modified:   openstack/compute/v2/servers/testing/requests_test.go

Unmerged paths:
  (use "git add <file>..." to mark resolution)
	both modified:   openstack/compute/v2/servers/requests.go

❯ vim openstack/compute/v2/servers/requests.go
❯ git add openstack/compute/v2/servers/requests.go
❯ git cherry-pick --continue
[v2 89180c87] Added support for hypervisor_hostname
 Date: Thu Feb 13 10:35:30 2025 +0900
 2 files changed, 28 insertions(+)
❯ git push origin v2 --force
Enumerating objects: 17, done.
Counting objects: 100% (17/17), done.
Delta compression using up to 24 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (9/9), 1.01 KiB | 1.01 MiB/s, done.
Total 9 (delta 7), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To github.com:ngin8-beta/gophercloud.git
 + 4f7baadc...89180c87 v2 -> v2 (forced update)
```

EmilienM commented on openstack/compute/v2/servers/requests.go:
```diff
@@ -508,6 +508,12 @@ type CreateOpts struct {
 	DiskConfig DiskConfig `json:"OS-DCF:diskConfig,omitempty"`
+
+	// KeyName is the name of the key pair.
+	KeyName string `json:"key_name,omitempty"`
```
this is from another PR and it was probably in a cherry-pick conflict.

EmilienM commented:
I resolved the conflict and pushed over your branch.

ngin8-beta commented:
thx!

ngin8-beta commented:
failure...😢

EmilienM commented:
> failure...😢

```
    volumeattach_test.go:21: Failure in volumeattach_test.go, line 21: unexpected error "No suitable endpoint could be found in the service catalog."
```

The error is not related.

@mandre are you fine backporting this?

ngin8-beta commented:
Will the check take a long time?
